hibernate-osgi/src/main/java/org/hibernate/osgi/HibernateBundleActivator.java

```diff
@@ -82,6 +84,14 @@ public void start(BundleContext context) throws Exception {
 		
 		sessionFactoryService = context.registerService( SessionFactory.class.getName(),
 				new OsgiSessionFactoryService( osgiClassLoader, osgiJtaPlatform, context ), new Hashtable());
+
+		context.addBundleListener(new BundleListener() {
+			@Override
+			public void bundleChanged(BundleEvent bundleEvent) {
+				if (bundleEvent.getType() == BundleEvent.UNRESOLVED)
```

Comment:
Is using `UNRESOLVED` enough?  Should we also remove the bundle from the CL if its stopped, then re-add once it's started?